hibernate-core/src/main/java/org/hibernate/query/sql/spi/ParameterRecognizer.java

```diff
@@ -49,4 +49,8 @@ public interface ParameterRecognizer {
 	default void complete() {
 		// by default, nothing to do
 	}
+
+	default boolean nativeJdbcParametersIgnored() {
```

Comment:
Actually, I think this should not be part of the `ParameterRecognizer` at all but rather part of the `NativeQueryInterpreterStandardImpl`. You can read the configuration in `org.hibernate.engine.query.spi.NativeQueryInterpreterInitiator` and pass that to the constructor of `NativeQueryInterpreterStandardImpl`. In `NativeQueryInterpreterStandardImpl#recognizeParameters`, you then call an overloaded `ParameterParser#parse` method that will additionally accept a boolean to indicate whether plain parameters should be parsed.

Reply:
`ParameterRecognizer` is a `callback` for collecting parameters if you read the Javadoc of it, so any sort of "configuration" simply doesn't belong to the interface. I guess it would be "ok" if the `ParameterRecognizerImpl` simply ignored calls to `ordinalParameter` when this flag is set, but the flag does not have to be exposed.